pkg/tickers/time.go

```diff
@@ -0,0 +1,74 @@
+package time
+
+import (
+	"context"
+	"log"
+	"time"
+
+	ocr2keepers "github.com/smartcontractkit/ocr2keepers/pkg"
+)
+
+var (
+	logPrintf = log.Printf
+)
+
+type observer interface {
+	Process(context.Context, Tick) error
+}
+
+// Ticker is a process that runs interval ticks.
+type Ticker interface {
+	Start() error
+	Stop() error
+}
+
+// Tick is the container for the individual tick
+type Tick interface {
+	// GetUpkeeps provides upkeeps scoped to the tick
+	GetUpkeeps(ctx context.Context) ([]ocr2keepers.UpkeepPayload, error)
+}
+
+type getUpkeepsFn func(context.Context, time.Time) (Tick, error)
+
+type timeTicker struct {
+	interval time.Duration
+	ticker   *time.Ticker
+	observer observer
+	getterFn getUpkeepsFn
+}
+
+func NewTimeTicker(interval time.Duration, observer observer, getterFn getUpkeepsFn) *timeTicker {
```

Comment:
For now using just the interval is fine, but there will probably be a case where the ticker interval is different from the `Process` timeout. The use case would be time ticks every second but an observe process that is allowed to take 5 seconds to complete. This would result in lost time intervals, but that is expected when `Process` blocks.

Reply:
Okay so should we pass an interval and a timeout duration here, or are we okay to leave it as-is for now?